### PR TITLE
feat: support systemd-boot ISO enroll keys option

### LINKED ISSUE #8196: Allow auto-enrollment of SecureBoot Keys into UEFI
## Feature Request

Currrently, the systemd-boot configuration is setup to only automatically enroll secure boot keys if it is booted in a VM:
https://github.com/siderolabs/talos/blob/main/pkg/imager/iso/loader.conf

In our environment we would like to set that option to `force` to always auto-enroll keys as long as the UEFI firmware is in setup mode, even when running bare-metal.

### PR DESCRIPTION
**Pull request by smira**
Fixes #8196

Example (profile excerpt):

```yaml
output:
  kind: iso
  isoOptions:
    sdBootEnrollKeys: force
  outFormat: raw
```

Defaults are still same (`if-safe` unless explicitly overridden).

### CODE REVIEW

File: pkg/imager/iso/uefi.go [hunk 1]
```diff
@@ -54,6 +58,8 @@ func CreateUEFI(printf func(string, ...any), options UEFIOptions) error {
 		return err
 	}
 
+	printf("preparing raw image")
```
Comment (frezbo) [inline pkg/imager/iso/uefi.go:61]: debug?

Comment (smira) [reply pkg/imager/iso/uefi.go:61]: nope, just some progress reporting ;)

File: pkg/imager/iso/uefi.go [hunk 2]
```diff
@@ -75,6 +81,18 @@ func CreateUEFI(printf func(string, ...any), options UEFIOptions) error {
...
 	}
 
+	printf("preparing loader.conf")
+
+	var loaderConfigOut bytes.Buffer
+
+	if err := template.Must(template.New("loader.conf").Parse(loaderConfigTemplate)).Execute(&loaderConfigOut, struct {
+		SecureBootEnroll string
+	}{
+		SecureBootEnroll: options.SDBootSecureBootEnrollKeys,
```
Comment (frezbo) [inline pkg/imager/iso/uefi.go:91]: should we do some validation of input? I don;t know how sd-boot behaves for an invalid key

Comment (smira) [reply pkg/imager/iso/uefi.go:91]: yep, I could add that, good point

**Review by frezbo — APPROVED** [review]

**Comment by frezbo** [comment]
maybe also something in the release notes?

File: pkg/imager/iso/uefi.go (hunk 2 above)
Comment (smira) [reply pkg/imager/iso/uefi.go:91]: I will actually rework it to use constants, this will be way better

**Review by frezbo — APPROVED** [review]

**Pull request merged**